projects/vdk-core/src/vdk/internal/builtin_plugins/builtin_hook_impl.py

```diff
@@ -58,31 +62,50 @@ def vdk_initialize(self, context: CoreContext) -> None:
 
         Configuration can override an attempt/op/execution id.
 
-        Otherwise each is auto-generated in format:
-        * Attempt id format is "execution_id-random_suffix"
-        * Execution id format is "op_id-random_suffix"
+        Attempt id and Execution id are extracted from env variables.
+        IF not present they are auto-generated.
+
+        If auto generated:
+        * Attempt id format is random string.
+        * Execution id is attempt id if not present in env.
         * Op Id is random string.
         """
-        op_id = context.configuration.get_value(CommonStoreKeys.OP_ID.key)
-        execution_id = context.configuration.get_value(CommonStoreKeys.EXECUTION_ID.key)
-        attempt_id = context.configuration.get_value(CommonStoreKeys.ATTEMPT_ID.key)
+        self._set_ids_from_env(context)
+        op_id = context.state.get(CommonStoreKeys.OP_ID)
+        execution_id = context.state.get(CommonStoreKeys.EXECUTION_ID)
+        attempt_id = context.state.get(CommonStoreKeys.ATTEMPT_ID)
 
-        if not op_id:
-            op_id = str(int(time.time()))
+        if not attempt_id:
+            attempt_id = f"{str(uuid.uuid4())[:6]}"
 
         if not execution_id:
-            execution_id = f"{op_id}-{str(uuid.uuid4())[:6]}"
+            execution_id = attempt_id
 
-        if not attempt_id:
-            attempt_id = f"{execution_id}-{str(uuid.uuid4())[:6]}"
+        if not op_id:
+            op_id = str(int(time.time()))
 
+        log.info(
+            f"Setting: OP_ID: {op_id}, ATTEMPT_ID: {attempt_id}, EXECUTION_ID: {execution_id}"
+        )
         context.state.set(CommonStoreKeys.OP_ID, op_id)
         context.state.set(CommonStoreKeys.EXECUTION_ID, execution_id)
         context.state.set(CommonStoreKeys.ATTEMPT_ID, attempt_id)
 
         context.state.set(CommonStoreKeys.VDK_VERSION, vdk_build_info.RELEASE_VERSION)
         context.state.set(CommonStoreKeys.START_TIME, datetime.utcnow())
 
+    def _set_ids_from_env(self, context: CoreContext):
+        """
+        Private method used to set the attempt id and
+        execution id from values present in the environment.
+        We are not setting  attempt id or execution id anywhere.
+        """
+        attempt_id = getenv(VDK_ATTEMPT_ID, None)
```

Comment:
Why need to use env variables?

use plugin configuration. The way plugin configuration works is explained if you type vdk config-help on top of it . There are different configuration providers. We have two:

One that reads from environment variables
One that reads from config.ini (with lower priority)
So we do not need to use getenv.

Reply:
```suggestion
        attempt_id = context.configuration.get_value(vdk_config.EXECUTION_ID)
```